src/functions_framework/__init__.py

```diff
@@ -68,18 +72,49 @@ def view_func(path):
     return view_func
 
 
-def _is_binary_cloud_event(request):
-    return (
-        request.headers.get("ce-type")
-        and request.headers.get("ce-specversion")
-        and request.headers.get("ce-source")
-        and request.headers.get("ce-id")
-    )
+def _run_cloudevent(function, request):
+    data = request.get_data()
+    event = from_http(data, request.headers)
+    function(event)
+
+
+def _cloudevent_view_func_wrapper(function, request):
+    def view_func(path):
+        try:
+            _run_cloudevent(function, request)
+        except (
+            cloud_exceptions.CloudEventMissingRequiredFields,
+            cloud_exceptions.CloudEventTypeErrorRequiredFields,
```

Comment:
This seems like two different errors, should we present them differently to the user?